Bug: https://trac.sagemath.org/ticket/33272

From 97d7958bed441cf2ccc714d88f83d3a8426bc085 Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Tue, 1 Feb 2022 23:18:20 +0100
Subject: Use roles when possible to fix warnings

---
 build/bin/sage-spkg-info                           |  4 ++--
 build/pkgs/ppl/SPKG.rst                            |  2 +-
 build/pkgs/thebe/SPKG.rst                          |  2 +-
 pkgs/sage-conf/README.rst                          |  4 ++--
 pkgs/sage-sws2rst/README.rst                       |  2 +-
 src/doc/en/developer/git_trac.rst                  |  2 +-
 src/doc/en/developer/packaging_sage_library.rst    |  2 +-
 src/doc/en/developer/trac.rst                      |  4 ++--
 src/doc/en/faq/faq-usage.rst                       |  2 +-
 src/doc/en/installation/source.rst                 | 24 +++++++++++-----------
 src/doc/en/reference/references/index.rst          | 24 +++++++++++-----------
 .../vector_calculus/vector_calc_advanced.rst       |  2 +-
 .../vector_calculus/vector_calc_curvilinear.rst    |  4 ++--
 .../vector_calculus/vector_calc_plane.rst          |  2 +-
 src/doc/it/faq/faq-usage.rst                       |  2 +-
 src/sage/categories/category_singleton.pyx         |  2 +-
 src/sage/coding/linear_rank_metric.py              |  2 +-
 src/sage/databases/oeis.py                         |  2 +-
 src/sage/graphs/generators/distance_regular.pyx    |  2 +-
 src/sage/knots/knotinfo.py                         |  2 +-
 src/sage/matrix/matrix_polynomial_dense.pyx        |  2 +-
 src/sage/modules/with_basis/invariant.py           |  2 +-
 src/sage/rings/function_field/order.py             |  2 +-
 src/sage/sat/solvers/satsolver.pyx                 |  2 +-
 24 files changed, 50 insertions(+), 50 deletions(-)

From c839fe705cf7656008928838bfacd370faf3dd40 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Mon, 21 Feb 2022 10:37:10 +1300
Subject: convert all links to trac to extlink during bootstrap

---
 src/doc/bootstrap | 2 ++
 1 file changed, 2 insertions(+)

From ab2365cfe215fc17f37a84f3bf04ad67b16667ad Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Mon, 21 Feb 2022 17:08:13 +1300
Subject: remove superfluous "ticket/"

---
 build/bin/sage-spkg-info                                              | 4 ++--
 build/pkgs/thebe/SPKG.rst                                             | 2 +-
 pkgs/sage-conf/README.rst                                             | 4 ++--
 pkgs/sage-sws2rst/README.rst                                          | 2 +-
 src/doc/bootstrap                                                     | 2 +-
 .../en/thematic_tutorials/vector_calculus/vector_calc_advanced.rst    | 3 +--
 src/doc/en/thematic_tutorials/vector_calculus/vector_calc_plane.rst   | 2 +-
 src/sage/matrix/matrix_polynomial_dense.pyx                           | 2 +-
 8 files changed, 10 insertions(+), 11 deletions(-)

From cedcffffaa432bdcdeb3f09447d311df1fec0b3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Mon, 21 Feb 2022 20:46:08 +1300
Subject: We don't need to touch any README.rst as they are not used to build
 the documentation

---
 pkgs/sage-conf/README.rst    | 4 ++--
 pkgs/sage-sws2rst/README.rst | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

From e8e80f326c3f49e5e6ffc4c3433bb5748f73e329 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Mon, 21 Feb 2022 20:47:04 +1300
Subject: Revert changes to `SPKG.rst`s, those changes can be applied when
 bootstraping the documentation

---
 build/pkgs/ppl/SPKG.rst   | 2 +-
 build/pkgs/thebe/SPKG.rst | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

From d5279466a35c7624a1f51f42236d9a76f0298493 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Mon, 21 Feb 2022 21:08:51 +1300
Subject: be more portable, do not use `sed -i`, pipe sed into the writing
 process.

---
 src/doc/bootstrap | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

From a29f7789301c291e1e28e2984c059b9c56ba8602 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Mon, 21 Feb 2022 21:54:17 +1300
Subject: reverting changes to sage-spkg-info

---
 build/bin/sage-spkg-info | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

From 1d8a60d6bcb30e48951b54e0ae8ec797d7b46c22 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Tue, 22 Feb 2022 09:33:29 +1300
Subject: removing some bracket that produce no interesting output and do not
 improve readibility.

---
 src/sage/matrix/matrix_polynomial_dense.pyx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

From 5251eb6fdc67339c0116ce6929acf1f0c9fd4551 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Tue, 22 Feb 2022 10:27:43 +1300
Subject: Update the formating of extlinks in accordance with
 https://sphinx-doc.org/en/master/usage/extensions/extlinks.html - this seem
 to be a sphinx 4.0 change.

---
 src/sage/misc/sagedoc.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

From 7230e35922abbc243c473de6d3c39afd9816f36d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Tue, 22 Feb 2022 13:16:36 +1300
Subject: fix broken link constructs - prototypes

---
 src/doc/en/developer/git_trac.rst | 2 +-
 src/doc/en/developer/trac.rst     | 4 ++--
 src/doc/en/faq/faq-usage.rst      | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

From 7d373f0e33bfecb1e5c6cd23bf8bb606b349a6e1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Tue, 22 Feb 2022 14:39:17 +1300
Subject: fix a forgotten "ticket/" entry

---
 src/doc/en/developer/packaging_sage_library.rst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

From 7c2766973cc44e071270af061c80391c8fa80116 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Tue, 22 Feb 2022 14:56:53 +1300
Subject: fix formatting in installation doc

---
 src/doc/en/installation/source.rst | 25 ++++++++++++-------------
 1 file changed, 12 insertions(+), 13 deletions(-)


From 17fa6d413eed1f1ecb7dd3b8076fc7d972b62d95 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Tue, 22 Feb 2022 20:15:02 +1300
Subject: fix the format of all remaining extlinks already touched on the
 branch.

---
 .../thematic_tutorials/vector_calculus/vector_calc_curvilinear.rst  | 6 ++----
 src/doc/it/faq/faq-usage.rst                                        | 3 +--
 src/sage/categories/category_singleton.pyx                          | 2 +-
 src/sage/coding/linear_rank_metric.py                               | 2 +-
 src/sage/knots/knotinfo.py                                          | 4 ++--
 src/sage/sat/solvers/satsolver.pyx                                  | 4 ++--
 6 files changed, 9 insertions(+), 12 deletions(-)

--- a/sage/src/doc/en/developer/git_trac.rst
+++ b/sage/src/doc/en/developer/git_trac.rst
@@ -108,7 +108,7 @@
 
 This is required if you authenticate to Trac with your GitHub account, as
 you do not have a Trac password.  Logged in users can find their token
-under `the token tab in preferences on the trac site <https://trac.sagemath.org/prefs/token>`_ .
+under :trac:`the token tab in preferences on the trac site <prefs/token>`.
 
 .. NOTE::
 
--- a/sage/src/doc/en/developer/packaging_sage_library.rst
+++ b/sage/src/doc/en/developer/packaging_sage_library.rst
@@ -204,7 +204,7 @@
 
 We can define some meaningful small distributions that just consist of
 a single or a few Cython modules. For example, **sagemath-tdlib**
-(https://trac.sagemath.org/ticket/29864) would just package the single
+(:trac:`29864`) would just package the single
 Cython module that must be linked with ``tdlib``,
 :mod:`sage.graphs.graph_decompositions.tdlib`. Starting with the Sage
 9.6 development cycle, as soon as namespace packages are activated, we
--- a/sage/src/doc/en/developer/trac.rst
+++ b/sage/src/doc/en/developer/trac.rst
@@ -17,7 +17,7 @@
 
 3. Link these two together.
 
-There is also a `wiki <https://trac.sagemath.org/wiki>`_ for more general
+There is also a :trac:`wiki <wiki>` for more general
 organizational web pages, like Sage development workshops.
 
 Thus if you find a bug in Sage, if you have new code to submit, want
@@ -25,7 +25,7 @@
 you have corrections for the documentation, you should post on the
 trac server. Items on the server are called *tickets*, and anyone may
 search or browse the tickets. For a list of recent changes, just visit
-the `Sage trac timeline <https://trac.sagemath.org/timeline>`_.
+the :trac:`Sage trac timeline <timeline>`.
 
 .. _section-trac-account:
 
--- a/sage/src/doc/en/faq/faq-usage.rst
+++ b/sage/src/doc/en/faq/faq-usage.rst
@@ -607,8 +607,8 @@
 
 produces the expected plot only for positive `x`. The *reason* is that Sage
 returns complex numbers for odd roots of negative numbers when numerically
-approximated, which is a `standard convention
-<https://en.wikipedia.org/wiki/Cube_root#Complex_numbers>`_. ::
+approximated, which is a 
+:wikipedia:`standard convention <Cube_root#Complex_numbers>`. ::
 
     sage: numerical_approx( (-1)^(1/3) )
     0.500000000000000 + 0.866025403784439*I
--- a/sage/src/doc/en/installation/source.rst
+++ b/sage/src/doc/en/installation/source.rst
@@ -15,7 +15,7 @@
    :depth: 2
 
 More familiarity with computers may be required to build Sage from
-the `source code <https://en.wikipedia.org/wiki/Source_code>`_.
+the :wikipedia:`source code <Source_code>`.
 If you do have all the :ref:`pre-requisite tools <section-prereqs>`,
 the process should be completely
 painless, basically consisting in extracting the source tarball and typing
@@ -43,13 +43,12 @@
 Supported platforms
 -------------------
 
-Sage runs on all major `Linux <https://en.wikipedia.org/wiki/Linux>`_
+Sage runs on all major :wikipedia:`Linux <Linux>`
 distributions, `macOS <https://www.apple.com/macosx/>`_ , and Windows
 (via the `Cygwin <https://cygwin.com/>`_ Linux API layer).
 
 Other installation options for Windows are using the Windows Subsystem
-for Linux (WSL), or with the aid of a `virtual machine
-<https://en.wikipedia.org/wiki/Virtual_machine>`_.
+for Linux (WSL), or with the aid of a :wikipedia:`virtual machine <Virtual_machine>`.
 
 .. _section-prereqs:
 
@@ -72,8 +71,8 @@
 Command-line tools
 ^^^^^^^^^^^^^^^^^^
 
-In addition to standard `POSIX <https://en.wikipedia.org/wiki/POSIX>`_ utilities
-and the `bash <https://en.wikipedia.org/wiki/Bash_(Unix_shell)>`_ shell,
+In addition to standard :wikipedia:`POSIX <POSIX>` utilities
+and the :wikipedia:`bash <Bash_(Unix_shell)>` shell,
 the following standard command-line development tools must be installed on your
 computer:
 
@@ -203,7 +202,7 @@
 
 on the command line. If it gives an error (or returns nothing), then
 either ``perl`` is not installed, or it is installed but not in your
-`PATH <https://en.wikipedia.org/wiki/PATH_%28variable%29>`_.
+:wikipedia:`PATH <PATH_%28variable%29>`.
 
 .. _sec-installation-from-sources-linux-recommended-installation:
 
@@ -224,7 +223,7 @@
 distribution, but on a `Debian <https://www.debian.org/>`_ based system (e.g.
 `Ubuntu <https://www.ubuntu.com/>`_ or `Mint <https://www.linuxmint.com/>`_),
 you would use
-`apt-get <https://en.wikipedia.org/wiki/Advanced_Packaging_Tool>`_.
+:wikipedia:`apt-get <Advanced_Packaging_Tool>`.
 
 On Debian ("buster" or newer) or Ubuntu ("bionic" or newer):
 
@@ -260,8 +259,8 @@
 .. literalinclude:: arch-optional.txt
 
 On other Linux systems, you might use
-`rpm <https://en.wikipedia.org/wiki/RPM_Package_Manager>`_,
-`yum <https://en.wikipedia.org/wiki/Yellowdog_Updater,_Modified>`_,
+:wikipedia:`rpm <RPM_Package_Manager>`,
+:wikipedia:`yum <Yellowdog_Updater,_Modified>`,
 or other package managers.
 
 .. _section_macprereqs:
@@ -589,7 +588,7 @@
 - **LaTeX**: highly recommended.
 
 It is highly recommended that you have
-`LaTeX <https://en.wikipedia.org/wiki/LaTeX>`_
+:wikipedia:`LaTeX <LaTeX>`
 installed, but it is not required.
 The most popular packaging is `TeX Live <https://www.tug.org/texlive/>`_,
 which can be installed following the directions on their web site.
@@ -908,7 +907,7 @@
 ------------
 
 To build Sage from scratch, you would typically execute ``make`` in Sage's home
-directory to build Sage and its `HTML <https://en.wikipedia.org/wiki/HTML>`_
+directory to build Sage and its :wikipedia:`HTML <HTML>`
 documentation.
 The ``make`` command is pretty smart, so if your build of Sage is interrupted,
 then running ``make`` again should cause it to pick up where it left off.
--- a/sage/src/doc/en/reference/references/index.rst
+++ b/sage/src/doc/en/reference/references/index.rst
@@ -632,7 +632,7 @@
 
 .. [Bec1992] Bernhard Beckermann. "A reliable method for computing M-Padé
              approximants on arbitrary staircases". J. Comput. Appl. Math.,
-             40(1):19-42, 1992. https://doi.org/10.1016/0377-0427(92)90039-Z.
+             40(1):19-42, 1992. :doi:`10.1016/0377-0427(92)90039-Z`.
 
 .. [BeCoMe] Frits Beukers, Henri Cohen, Anton Mellit,
    *Finite hypergeometric functions*,
@@ -870,7 +870,7 @@
 .. [BL1994] Bernhard Beckermann, George Labahn. "A Uniform Approach for the
             Fast Computation of Matrix-Type Padé Approximants". SIAM J. Matrix
             Anal. Appl. 15 (1994) 804-823.
-            http://dx.doi.org/10.1137/S0895479892230031
+            :doi:`10.1137/S0895479892230031`
 
 .. [BMP2007] \S. Brlek, G. Melançon, G. Paquin, Properties of the
              extremal infinite smooth words, Discrete
@@ -915,11 +915,11 @@
 
 .. [BLV1999] Bernhard Beckermann, George Labahn, and Gilles Villard. "Shifted
              normal forms of polynomial matrices". In ISSAC'99, pages 189-196.
-             ACM, 1999. https://doi.org/10.1145/309831.309929 .
+             ACM, 1999. :doi:`10.1145/309831.309929`.
 
 .. [BLV2006] Bernhard Beckermann, George Labahn, and Gilles Villard. "Normal
              forms for general polynomial matrices". J. Symbolic Comput.,
-             41(6):708-737, 2006. https://doi.org/10.1016/j.jsc.2006.02.001 .
+             41(6):708-737, 2006. :doi:`10.1016/j.jsc.2006.02.001`.
 
 .. [BM1940] Becker, M. F., and Saunders MacLane. The minimum number of
             generators for inseparable algebraic extensions. Bulletin of the
@@ -1416,7 +1416,7 @@
                Graphs*. In: Klasing R. (eds) Experimental Algorithms. SEA 2012.
                Lecture Notes in Computer Science, vol 7276.
                Springer, Berlin, Heidelberg
-               :doi:`https://doi.org/10.1007/978-3-642-30850-5_10`.
+               :doi:`10.1007/978-3-642-30850-5_10`.
 
 .. [CGW2013] Daniel Cabarcas, Florian Göpfert, and Patrick
              Weiden. Provably Secure LWE-Encryption with Uniform
@@ -2015,7 +2015,7 @@
 .. [DPS2017] Kevin Dilks, Oliver Pechenik, and Jessica Striker,
              *Resonance in orbits of plane partitions and increasing
              tableaux*, JCTA 148 (2017), 244-274,
-             https://doi.org/10.1016/j.jcta.2016.12.007
+             :doi:`10.1016/j.jcta.2016.12.007`
 
 .. [DPV2001] \J. Daemen, M. Peeters, and G. Van Assche,
              *Bitslice ciphers and power analysis attacks*; in FSE, (2000), pp. 134-149.
@@ -2048,7 +2048,7 @@
             Volume 69, Issue 4,
             2004,
             Pages 525-546,
-            https://doi.org/10.1016/j.jcss.2004.03.004.
+            :doi:`10.1016/j.jcss.2004.03.004`.
 
 .. [DS2009] \W. Decker, F.-O. Schreyer,
             *Varieties, Gröbner Bases, and Algebraic Curves*, 2009.
@@ -2481,7 +2481,7 @@
              file format, Mathematical software – ICMS 2016. 5th international
              congress, Berlin, Germany, July 11–14, 2016. Proceedings, Berlin:
              Springer, 2016, pp. 403–410,
-             https://doi.org/10.1007/978-3-319-42432-3_50, ISBN
+             :doi:`10.1007/978-3-319-42432-3_50`, ISBN
              978-3-319-42431-6/pbk.
 
 .. [GHJV1994] \E. Gamma, R. Helm, R. Johnson, J. Vlissides, *Design
@@ -2507,7 +2507,7 @@
 .. [GJ2006] Ewgenij Gawrilow and Michael Joswig, Flexible object hierarchies in
             polymake (extended abstract), Mathematical software—ICMS 2006,
             Lecture Notes in Comput. Sci., vol. 4151, Springer, Berlin, 2006,
-            pp. 219–221, https://doi.org/10.1007/11832225_20
+            pp. 219–221, :doi:`10.1007/11832225_20`
 
 .. [GJ2007] \A. Glen, J. Justin, Episturmian words: a survey, Preprint,
             2007, :arxiv:`0801.1655`.
@@ -2533,7 +2533,7 @@
 
 .. [GJRW2010] Ewgenij Gawrilow, Michael Joswig, Thilo Rörig, and Nikolaus Witte,
               Drawing polytopal graphs with polymake, Comput. Vis. Sci. 13
-              (2010), no. 2, 99–110, https://doi.org/10.1007/s00791-009-0127-3
+              (2010), no. 2, 99–110, :doi:`10.1007/s00791-009-0127-3`
 
 .. [GK1982] Daniel H. Greene and Donald E. Knuth (1982), "2.1.1 Constant
             coefficients - A) Homogeneous equations", Mathematics for the Analysis
@@ -2583,7 +2583,7 @@
 
 .. [GN2018] Pascal Giorgi and Vincent Neiger. Certification of Minimal
             Approximant Bases. In ISSAC 2018, pages 167-174.
-            https://doi.org/10.1145/3208976.3208991
+            :doi:`10.1145/3208976.3208991`
 
 .. [Go1967] Solomon Golomb, Shift register sequences, Aegean Park
             Press, Laguna Hills, Ca, 1967
@@ -3219,7 +3219,7 @@
 .. [JNSV2016] Claude-Pierre Jeannerod, Vincent Neiger, Eric Schost, and Gilles
              Villard. Fast Computation of Minimal Interpolation Bases in Popov
              Form for Arbitrary Shifts. In Proceedings ISSAC 2016 (pages
-             295-302). https://doi.org/10.1145/2930889.2930928
+             295-302). :doi:`10.1145/2930889.2930928`
 
 .. [Joh1990] \D.L. Johnson. *Presentations of Groups*. Cambridge
              University Press. (1990).
--- a/sage/src/doc/en/thematic_tutorials/vector_calculus/vector_calc_advanced.rst
+++ b/sage/src/doc/en/thematic_tutorials/vector_calculus/vector_calc_advanced.rst
@@ -39,8 +39,7 @@
     Riemannian metric g on the Euclidean space E^3
 
 Actually ``RR`` is used here as a proxy for the real field (this should be
-replaced in the future, see the discussion at `#24456
-<https://trac.sagemath.org/ticket/24456>`__) and the 53 bits of precision play
+replaced in the future, see the discussion at :trac:`24456`) and the 53 bits of precision play
 of course no role for the symbolic computations.
 
 Let us introduce spherical and cylindrical coordinates on
--- a/sage/src/doc/en/thematic_tutorials/vector_calculus/vector_calc_curvilinear.rst
+++ b/sage/src/doc/en/thematic_tutorials/vector_calculus/vector_calc_curvilinear.rst
@@ -277,8 +277,7 @@
      + d^2(u_phi)/dph^2/(r^2*sin(th)^2) + d^2(u_phi)/dr^2
 
 As a test, we may check that these formulas coincide with those of Wikipedia's
-article `Del in cylindrical and spherical coordinates
-<https://en.wikipedia.org/wiki/Del_in_cylindrical_and_spherical_coordinates#Del_formula>`__.
+article :wikipedia:`Del in cylindrical and spherical coordinates <Del_in_cylindrical_and_spherical_coordinates#Del_formula>`.
 
 Using cylindrical coordinates
 -----------------------------
@@ -392,8 +391,7 @@
     Delta(u)^3 = d(u_z)/drh/rh + d^2(u_z)/dph^2/rh^2 + d^2(u_z)/drh^2 + d^2(u_z)/dz^2
 
 Again, we may check that the above formulas coincide with those of Wikipedia's
-article `Del in cylindrical and spherical coordinates
-<https://en.wikipedia.org/wiki/Del_in_cylindrical_and_spherical_coordinates#Del_formula>`__.
+article :wikipedia:`Del in cylindrical and spherical coordinates <Del_in_cylindrical_and_spherical_coordinates#Del_formula>`.
 
 Changing coordinates
 --------------------
--- a/sage/src/doc/en/thematic_tutorials/vector_calculus/vector_calc_plane.rst
+++ b/sage/src/doc/en/thematic_tutorials/vector_calculus/vector_calc_plane.rst
@@ -500,7 +500,7 @@
 
 Actually ``RR`` is used here as a proxy for the real field (this should
 be replaced in the future, see the discussion at
-`#24456 <https://trac.sagemath.org/ticket/24456>`__) and the 53 bits of
+:trac:`24456`) and the 53 bits of
 precision play of course no role for the symbolic computations.
 
 The user atlas of `\mathbb{E}^2` has two charts::
--- a/sage/src/doc/it/faq/faq-usage.rst
+++ b/sage/src/doc/it/faq/faq-usage.rst
@@ -590,8 +590,7 @@
 produce il grafico corretto solo per valori di `x` positivi. La
 *ragione* per cui ciò avviene è che Sage restituisce dei numeri
 complessi per le radici dispari di numeri negativi, quando queste sono
-approssimate, il che è una `convenzione standard
-<https://en.wikipedia.org/wiki/Cube_root#Complex_numbers>`_::
+approssimate, il che è una :wikipedia:`convenzione standard <Cube_root#Complex_numbers>`::
 
     sage: numerical_approx( (-1)^(1/3) )
     0.500000000000000 + 0.866025403784439*I
--- a/sage/src/sage/categories/category_singleton.pyx
+++ b/sage/src/sage/categories/category_singleton.pyx
@@ -89,7 +89,7 @@
 
     A *singleton* category is a category whose class takes no
     parameters like ``Fields()`` or ``Rings()``. See also the
-    `Singleton design pattern <https://en.wikipedia.org/wiki/Singleton_pattern>`_.
+    :wikipedia:`Singleton design pattern <Singleton_pattern>`.
 
     This is a subclass of :class:`Category`, with a couple
     optimizations for singleton categories.
--- a/sage/src/sage/coding/linear_rank_metric.py
+++ b/sage/src/sage/coding/linear_rank_metric.py
@@ -86,7 +86,7 @@
 ------------------
 
 Read more about
-`rank metric and Gabidulin codes <https://en.wikipedia.org/wiki/Rank_error-correcting_code>`_
+:wikipedia:`rank metric and Gabidulin codes <Rank_error-correcting_code>`
 
 AUTHORS:
 
--- a/sage/src/sage/databases/oeis.py
+++ b/sage/src/sage/databases/oeis.py
@@ -1023,7 +1023,7 @@
         .. TODO::
 
             - ask OEIS to add a keyword telling whether the sequence comes from
-              a power series, e.g. for https://oeis.org/A000182
+              a power series, e.g. for :oeis:`A000182`
             - discover other possible conversions.
 
         EXAMPLES::
--- a/sage/src/sage/graphs/generators/distance_regular.pyx
+++ b/sage/src/sage/graphs/generators/distance_regular.pyx
@@ -985,7 +985,7 @@
 
     See [BCN1989]_ pp. 264, 265 or [VDKT2016]_ p. 21.
     This construction can be found on
-    https://en.wikipedia.org/wiki/Halved_cube_graph#Equivalent_constructions
+    :wikipedia:`Halved_cube_graph#Equivalent_constructions`
 
     TESTS:
 
--- a/sage/src/sage/knots/knotinfo.py
+++ b/sage/src/sage/knots/knotinfo.py
@@ -41,8 +41,8 @@
 Also, note that the braid notation is used according to Sage, even thought in
 the source where it is taken from, the braid generators are assumed to have a
 negative crossing which would be opposite to the convention in Sage (see definition
-3 of `Gittings, T., "Minimum Braids: A Complete Invariant of Knots and Links
-<https://arxiv.org/abs/math/0401051>`__).
+3 of
+:arxiv:`Gittings, T., "Minimum Braids: A Complete Invariant of Knots and Links" <math/0401051>`).
 
 For different conventions regarding normalization of the polynomial invariants see
 the according documentation of :meth:`KnotInfoBase.homfly_polynomial`,
--- a/sage/src/sage/matrix/matrix_polynomial_dense.pyx
+++ b/sage/src/sage/matrix/matrix_polynomial_dense.pyx
@@ -781,7 +781,7 @@
             2021), it would be highly beneficial to use conversions and rely on
             polynomials with matrix coefficients when the matrix size is
             "large" and the degree "small", see
-            [https://trac.sagemath.org/ticket/31472#comment:5].
+            :trac:`31472#comment:5`.
         """
         if d <= 0:
             raise ValueError("the precision must be positive")
--- a/sage/src/sage/modules/with_basis/invariant.py
+++ b/sage/src/sage/modules/with_basis/invariant.py
@@ -205,7 +205,7 @@
         Extend when `M` does not have a basis and `S` is a permutation
         group using:
 
-        - https://arxiv.org/abs/0812.3082
+        - :arxiv:`0812.3082`
         - https://www.dmtcs.org/pdfpapers/dmAA0123.pdf
     """
     def __init__(self, M, S, action=operator.mul, side='left', *args, **kwargs):
--- a/sage/src/sage/rings/function_field/order.py
+++ b/sage/src/sage/rings/function_field/order.py
@@ -1772,7 +1772,7 @@
         finite-dimensional algebra, a construct for which we do
         support primary decomposition.
 
-        See https://trac.sagemath.org/attachment/ticket/28094/decomposition.pdf
+        See :trac:`attachment/ticket/28094/decomposition.pdf`
 
         .. TODO::
 
--- a/sage/src/sage/sat/solvers/satsolver.pyx
+++ b/sage/src/sage/sat/solvers/satsolver.pyx
@@ -314,8 +314,8 @@
     r"""
     Return a :class:`SatSolver` instance.
 
-    Through this class, one can define and solve `SAT
-    <https://en.wikipedia.org/wiki/Boolean_satisfiability_problem>`__ problems.
+    Through this class, one can define and solve
+    :wikipedia:`SAT problems <Boolean_satisfiability_problem>`.
 
     INPUT:
 
--- a/sage/src/doc/bootstrap
+++ b/sage/src/doc/bootstrap
@@ -171,8 +171,9 @@
             PKG_SCRIPTS=build/pkgs/$PKG_BASE
             # Instead of just copying, we may want to call
             # a version of sage-spkg-info to format extra information.
+            # for sphinx 4.4 we need to replace all direct links by some "extlink" (ticket 33272)
 
-            (echo ".. _spkg_$PKG_BASE:" && echo && OUTPUT_RST=1 sage-spkg-info $PKG_BASE) > "$OUTPUT_DIR"/$PKG_BASE.rst
+            (echo ".. _spkg_$PKG_BASE:" && echo && OUTPUT_RST=1 sage-spkg-info $PKG_BASE) | sed -e "s|https://trac.sagemath.org/ticket/\([0-9]*\)|:trac:\`\1\`|g" -e "s|https://arxiv.org/abs/cs/\([0-9]*\)|:arxiv:\`cs/\1\`|g" > "$OUTPUT_DIR"/$PKG_BASE.rst
             echo >> "$OUTPUT_INDEX" "   $PKG_BASE"
 done
 
--- a/sage/src/sage/misc/sagedoc.py
+++ b/sage/src/sage/misc/sagedoc.py
@@ -495,13 +495,13 @@
 pythonversion = sys.version.split(' ')[0]
 extlinks = {
     'python': ('https://docs.python.org/release/'+pythonversion+'/%s', ''),
-    'trac': ('https://trac.sagemath.org/%s', 'trac ticket #'),
-    'wikipedia': ('https://en.wikipedia.org/wiki/%s', 'Wikipedia article '),
-    'arxiv': ('https://arxiv.org/abs/%s', 'arXiv '),
-    'oeis': ('https://oeis.org/%s', 'OEIS sequence '),
-    'doi': ('https://doi.org/%s', 'doi:'),
-    'pari': ('https://pari.math.u-bordeaux.fr/dochtml/help/%s', 'pari:'),
-    'mathscinet': ('https://www.ams.org/mathscinet-getitem?mr=%s', 'MathSciNet ')
+    'trac': ('https://trac.sagemath.org/%s', 'trac ticket #%s'),
+    'wikipedia': ('https://en.wikipedia.org/wiki/%s', 'Wikipedia article %s'),
+    'arxiv': ('https://arxiv.org/abs/%s', 'arXiv %s'),
+    'oeis': ('https://oeis.org/%s', 'OEIS sequence %s'),
+    'doi': ('https://doi.org/%s', 'doi:%s'),
+    'pari': ('https://pari.math.u-bordeaux.fr/dochtml/help/%s', 'pari:%s'),
+    'mathscinet': ('https://www.ams.org/mathscinet-getitem?mr=%s', 'MathSciNet %s')
 }
 
 def process_extlinks(s, embedded=False):
